### PR TITLE
build: align with internal tsconfig options

### PR DESCRIPTION
**Pull request by crisbeto**
Currently internally Angular has some customized tsconfig files, because we don't align with the tsconfig of the rest of g3. These changes enable `noImplicitReturns` and `noPropertyAccessFromIndexSignature` to align better with the internal config.

### CODE REVIEW

File: aio/.eslintrc.json [hunk 1]
```diff
@@ -120,7 +120,6 @@
         "complexity": "off",
         "constructor-super": "error",
         "curly": "error",
-        "dot-notation": "error",
```
Comment (crisbeto) [inline aio/.eslintrc.json:123]: Context here: there were two different rules checking the dot notation: this one and `@typescript-eslint/dot-notation`.

**Review by devversion — APPROVED** [review]
LGTM. Great cleanup! - (linking to the CL for reference: http://cl/563539564)

**Review by AndrewKushnir — APPROVED** [review]
LGTM 👍 

Note: this PR will require a Global Presubmit (TGP) to make sure that `.foo` -> `['foo']` changes are safe with Closure optimizations.

**Review by jelbourn — APPROVED** [review]
LGTM

**Review by alxhub — APPROVED** [review]
Reviewed-for: global-approvers

**Comment by crisbeto** [comment]
[Passing TGP](http://test/OCL:564600909:BASE:564626339:1694503032734:c6cc7fa9)

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 52cc7f839b67b64d8deb0730b11435f3e0a24444.

**Pull request closed without merging**